### PR TITLE
Reflect newly introduced funnel to count rejecting reasons in the api

### PR DESCRIPTION
**Pull request by unterstein**
Enhanced declined offers `/v2/queue` api to return a triple (reason, amount declined, amount processed) for the rejectSummaryLastOffers property

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/mesosphere/marathon/pull/4674" target="_blank">ReviewNinja

File: src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala [hunk 1]
```diff
@@ -26,14 +26,24 @@ trait QueueInfoConversion extends DefaultConversions with OfferConversion {
...
-        unusedOffersCount = info.unusedOffersCount,
-        lastUnusedOfferAt = info.lastNoMatch.map(_.timestamp.toOffsetDateTime),
-        lastUsedOfferAt = info.lastMatch.map(_.timestamp.toOffsetDateTime),
-        rejectSummaryLastOffers = Raml.toRaml(info.rejectSummaryLastOffers),
-        rejectSummaryLaunchAttempt = Raml.toRaml(info.rejectSummaryLaunchAttempt)
-      )
+      def processedOffersSummary: ProcessedOffersSummary = {
+        // `rejectSummaryLastOffers` should be a triple of (reason, amount declined, amount processed)
+        // and should reflect the `NoOfferMatchReason.reasonFunnel` to store only first non matching reason.
+        val (_, rejectSummaryLastOffers) = NoOfferMatchReason.
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala]: make this a function: Map[NoOfferMatch, Int] => Seq[RejectOfferStep]

File: docs/docs/rest-api/public/api/v2/types/queue.raml [hunk 2]
```diff
@@ -41,7 +52,7 @@ types:
         type: datetime
         description: Point in time when the last used offer has been processed.
       rejectSummaryLastOffers?:
```
Comment (aquamatthias) [inline docs/docs/rest-api/public/api/v2/types/queue.raml]: Also compute rejectSummaryLaunchAttempt the same way as  rejectSummaryLastOffers

File: src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala [hunk 3]
```diff
@@ -39,6 +39,7 @@ object LaunchQueue {
     rejectSummaryLastOffers: Map[NoOfferMatchReason, Int],
     rejectSummaryLaunchAttempt: Map[NoOfferMatchReason, Int],
     processedOffersCount: Int,
+    lastProcessedOffersCount: Int,
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala]: why? you get all lastNoMatches

File: src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala [hunk 4]
```diff
@@ -60,6 +60,7 @@ class OfferMatchStatisticsActor extends Actor {
         lastOfferSummary(lastOffers),
         statistics.rejectSummary,
         statistics.processedOfferCount,
+        lastOffers.size,
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala]: Not needed. See other comment.

File: src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala (hunk 3 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala]: !resolve

File: docs/docs/rest-api/public/api/v2/types/queue.raml (hunk 2 above)
Comment (aquamatthias) [inline docs/docs/rest-api/public/api/v2/types/queue.raml]: !fix

File: src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala (hunk 4 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala]: !resolve

File: src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala (hunk 1 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala]: !resolve

File: src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala (hunk 3 above)
Comment (unterstein) [inline src/main/scala/mesosphere/marathon/core/launchqueue/LaunchQueue.scala]: You are so right 👀 !resolved

File: src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala (hunk 4 above)
Comment (unterstein) [inline src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala]: 👀 !resolved

File: docs/docs/rest-api/public/api/v2/types/queue.raml [hunk 5]
```diff
@@ -24,6 +24,17 @@ types:
         type: number
         format: int32
 
+  LastOfferRejectionSummary:
```
Comment (unterstein) [inline docs/docs/rest-api/public/api/v2/types/queue.raml]: !fixed

File: docs/docs/rest-api/public/api/v2/types/queue.raml (hunk 2 above)
Comment (unterstein) [inline docs/docs/rest-api/public/api/v2/types/queue.raml]: !fixed

**Comment by unterstein** [comment]
retest this please

**Comment by aquamatthias** [comment]
Thanks!

File: src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala [hunk 6]
```diff
@@ -26,14 +26,34 @@ trait QueueInfoConversion extends DefaultConversions with OfferConversion {
...
 
-      def processedOffersSummary: ProcessedOffersSummary = ProcessedOffersSummary(
-        processedOffersCount = info.processedOffersCount,
-        unusedOffersCount = info.unusedOffersCount,
-        lastUnusedOfferAt = info.lastNoMatch.map(_.timestamp.toOffsetDateTime),
-        lastUsedOfferAt = info.lastMatch.map(_.timestamp.toOffsetDateTime),
-        rejectSummaryLastOffers = Raml.toRaml(info.rejectSummaryLastOffers),
-        rejectSummaryLaunchAttempt = Raml.toRaml(info.rejectSummaryLaunchAttempt)
-      )
+      /**
```
Comment (unterstein) [inline src/main/scala/mesosphere/marathon/raml/QueueInfoConversion.scala]: Thanks, will fix this soon

**Pull request merged**